Development/nmos/video_h264.h

```diff
@@ -0,0 +1,116 @@
+#ifndef NMOS_VIDEO_H264_H
+#define NMOS_VIDEO_H264_H
+
+#include "nmos/media_type.h"
+#include "nmos/sdp_utils.h"
+
+namespace sdp
+{
+    // Packetization Mode
+    // See https://www.iana.org/assignments/media-types/video/H264
+    // and https://tools.ietf.org/html/rfc6184#section-6
+    enum packetization_mode
+    {
+        // Single NAL Unit Mode
+        single_nal_unit_mode = 0,
+        // Non-Interleaved Mode
+        non_interleaved_mode = 1,
+        // Interleaved Mode
+        interleaved_mode = 2
+    };
+
+    namespace fields
+    {
+        // See https://www.iana.org/assignments/media-types/video/H264
+        // and https://tools.ietf.org/html/rfc6184#section-6
+        const web::json::field_as_string_or profile_level_id{ U("profile-level-id"), {} }; // hex
+        const web::json::field_with_default<uint32_t> packetization_mode{ U("packetization-mode"), (uint32_t)single_nal_unit_mode };
+        const web::json::field_as_string_or sprop_parameter_sets{ U("sprop-parameter-sets"), {} };
+    }
+}
+
+namespace nmos
+{
+    namespace media_types
+    {
+        // H.264 Video
+        // See https://www.iana.org/assignments/media-types/video/H264
+        // and https://tools.ietf.org/html/rfc6184
+        const media_type video_H264{ U("video/H264") };
+    }
+
+    // Packetization Mode
+    // See [TBC in NMOS Parameter Registers]
+    // and https://www.iana.org/assignments/media-types/video/H264
+    // and https://tools.ietf.org/html/rfc6184#section-6
+    DEFINE_STRING_ENUM(packetization_mode)
+    namespace packetization_modes
+    {
+        // Single NAL Unit Mode
+        const packetization_mode single_nal_unit_mode{ U("single_nal_unit_mode") };
+        // Non-Interleaved Mode
+        const packetization_mode non_interleaved_mode{ U("non_interleaved_mode") };
+        // Interleaved Mode
+        const packetization_mode interleaved_mode{ U("interleaved_mode") };
+    }
+
+    sdp::packetization_mode make_sdp_packetization_mode(const packetization_mode& packetization_mode);
+    packetization_mode parse_sdp_packetization_mode(const sdp::packetization_mode& packetization_mode);
+
+    namespace fields
+    {
+        // See [TBC in NMOS Parameter Registers]
+
+        // flow_video_coded
+        const web::json::field<uint32_t> profile_level_id{ U("profile_level_id") };
+        // sender
+        const web::json::field_as_string packetization_mode{ U("packetization_mode") };
+    }
+
+    namespace caps
+    {
+        // See [TBC in NMOS Parameter Registers]
+
+        namespace format
+        {
+            const web::json::field_as_value_or profile_level_id{ U("urn:x-nmos:cap:format:profile_level_id"), {} }; // number
+        }
+
+        namespace transport
+        {
+            const web::json::field_as_value_or packetization_mode{ U("urn:x-nmos:cap:transport:packetization_mode"), {} }; // number
```

Comment:
Actually at the moment in the _nmos-cpp-node/node_implementation.cpp_ example I've put the string value in, same as the Flow property. We should discuss again whether it's a good idea to have numeric values or strings for these in NMOS.